packages/main/src/plugin/index.spec.ts

```diff
@@ -59,26 +78,68 @@ webContents.send = vi.fn();
 
 const mainWindowDeferred = new Deferred<BrowserWindow>();
 
+vi.mock('/@/plugin/tasks/task-manager.js', () => ({
+  TaskManager: vi.fn(),
+}));
+
+vi.mock('/@/plugin/updater.js', () => ({
+  Updater: vi.fn(),
+}));
+
+vi.mock('./extension-loader.js', () => ({
+  ExtensionLoader: vi.fn(),
+}));
+
+vi.mock('./container-registry.js', () => ({
+  ContainerProviderRegistry: vi.fn(),
+}));
+
+vi.mock('electron', () => {
+  return {
+    shell: {
+      openExternal: vi.fn(),
+    },
+    app: {
+      on: vi.fn(),
+      getVersion: vi.fn(),
+    },
+    clipboard: {
+      writeText: vi.fn(),
+    },
+    BrowserWindow: {
+      getAllWindows: vi.fn(),
+    },
+    ipcMain: {
+      on: vi.fn(),
+      handle: vi.fn(),
+    },
+  };
+});
+
 beforeAll(() => {
-  vi.mock('electron', () => {
-    return {
-      shell: {
-        openExternal: vi.fn(),
-      },
-      app: {
-        on: vi.fn(),
-      },
-      clipboard: {
-        writeText: vi.fn(),
-      },
-    };
-  });
   const trayMenuMock = {} as unknown as TrayMenu;
   pluginSystem = new TestPluginSystem(trayMenuMock, mainWindowDeferred);
 });
 
+const MAIN_WINDOW_MOCK: BrowserWindow = {
+  isDestroyed: vi.fn(),
+  webContents: {
+    on: vi.fn(),
+  },
+} as unknown as BrowserWindow;
+
 beforeEach(() => {
   vi.clearAllMocks();
+
+  vi.mocked(BrowserWindow.getAllWindows).mockReturnValue([MAIN_WINDOW_MOCK]);
```

Comment:
issue: I think this pattern is not expected. I have never seen it either in official vitest documentation or in our tests.

it's really hard to then understand what is being done, as the mock declared in vi.mock is not used at all.

Reply:
The mock in vi.mock is needed otherwise when accessing `BrowserWindow.getAllWindows` we would get a `not a mocked object`.

I feel like there is nothing new here ? How do you mock for all the tests the return value of a function ? Here I want `getAllWindows` to return the MAIN_WINDOW_MOCK for all the tests. If I do the following

```
vi.mock('electron', () => {
  return {
    BrowserWindow: {
      getAllWindows: vi.fn().mockReturnValue([MAIN_WINDOW_MOCK]),
    },
});
```

The `vi.clearAllMocks()` will remove the mock return value for my tests. So I need to mock the return value in the beforeEach.

Reply:
there is this one and these :
```
    vi.mocked(TaskManager).mockReturnValue(TASK_MANAGER_MOCK);
    vi.mocked(ContainerProviderRegistry).mockReturnValue(CONTAINER_PROVIDER_REGISTRY);
    vi.mocked(CONTAINER_PROVIDER_REGISTRY.buildImage).mockResolvedValue(undefined);
```

which is also related to https://github.com/containers/podman-desktop-extension-ai-lab/pull/2345

it looks like you're pushing a new pattern which is not documented or widely used.

Reply:
Removed all non-standard way of testing (masking as resolved as I changed all the tests)